libknet/compress.c

```diff
@@ -209,6 +209,28 @@ static int compress_lib_test(knet_handle_t knet_h)
 		log_err(knet_h, KNET_SUB_COMPRESS, "Unable to compress test buffer. Please check your compression settings: %s", strerror(savederrno));
 		errno = savederrno;
 		return -1;
+	} else if ((long unsigned int)dst_comp_len >= KNET_DATABUFSIZE) {
+		/*
+		 * compress not effect, try again using default compression level
+		 */
+		log_warn(knet_h, KNET_SUB_COMPRESS, "Useless compression level: %d", knet_h->compress_level);
+
+		if ((!compress_modules_cmds[knet_h->compress_model].ops->get_default_level()) ||
+				(knet_h->compress_level = compress_modules_cmds[knet_h->compress_model].ops->get_default_level() < 0)) {
+                        log_err(knet_h, KNET_SUB_COMPRESS, "Get default compression level failed");
```

Comment:
Please expand the message a bit here.
"compression %s does not provide a default value", compress_module_name....